cmd/scale.go

```diff
@@ -375,8 +375,10 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 
 	addValue(parametersJSON, sc.agentPool.Name+"Count", countForTemplate)
 
+	// The agent pool is set to index 0 for the scale operation, we need to overwrite the template variables that rely on pool index.
 	if winPoolIndex != -1 {
 		templateJSON["variables"].(map[string]interface{})[sc.agentPool.Name+"Index"] = winPoolIndex
+		templateJSON["variables"].(map[string]interface{})[sc.agentPool.Name+"VMNamePrefix"] = sc.containerService.Properties.GetAgentVMPrefix(sc.agentPool, winPoolIndex)
```

Comment:
FYI for reviewers I verified that this doesn't affect AKS since the `GetAgentVMPrefix` has a special case to handle the `v2` naming: we're just recalling the same function as the template generator here, not hardcoding the name.